### PR TITLE
Adds tests cases for issue #5702

### PR DESCRIPTION
**Pull request by peetzweg**
See #5702 for further details.

### CODE REVIEW

**Review by jacogr — CHANGES_REQUESTED** [review]
Always appreciate this. Obviously it cannot be merged but useful for somebody that wishes to look into the logged issue - if CI fails the PR is not merge-able, but somebody can certainly copy this/parts of it into their effort.

One comment:

If anything fails you need to delve through an it statement to find the place - so each `it` should have 1 expectation (maybe more, if they are very closely related and follow-ups of each other and easy to find)

In some cases it is possible, in others it would be impossible to find which one creates issues, for instance

```js
// Zero
    expect(
      new Int(registry, new Uint8Array([0]), 8).toNumber()
    ).toEqual(0);

    expect(
      new Int(registry, new Uint8Array([0, 0]), 16).toNumber()
    ).toEqual(0);
```

TL;DR Rather use a `describe` block with specific `it` + (one) `expect` for each.

**Comment by peetzweg** [comment]
Thanks @jacogr for pointing out where these errors might root from. Tried looking into it a bit already, will continue from there.

> In some cases it is possible, in others it would be impossible to find which one creates issues, for instance

Yes, agree it's not as transparent which one failed, however the stacktrace reveals which one it actually failed i.e.
```
  at TestContext.<anonymous> (/Users/xxx/git/pjs-api/packages/types-codec/src/base/Int.spec.ts:113:7)
``` 
  
Copy pasting almost identical `it`s is quite cumbersome. Unfortunately we can't create [parameterized tests](https://vitest.dev/api/#test-each) with @polkadot/dev-test right?

There also seems to be an issue with decoding just the value `0` to a Number, which results in 
```
i8: Input too large. Found input with 9 bits, expected 8
```
https://github.com/peetzweg/api/blob/f33bbc8340d66f46b77da041bfc1467b3d33c450/packages/types-codec/src/base/Int.spec.ts#L111-L114

Which is a quite odd. 🤔

**Comment by jacogr** [comment]
Could you merge in the master branch (it contains the latest common stuff). Hopefully it means the test cases here would start passing…

**Comment by peetzweg** [comment]
@jacogr done! They succeed locally now. Great job! 🙌

**Review by jacogr — DISMISSED** [review]

File: packages/types-codec/src/base/Int.spec.ts [hunk 1]
```diff
@@ -41,6 +41,108 @@ describe('Int', (): void => {
...
+  it('decodes edge case to js number', (): void => {
+    // Zero
+    expect(
+      new Int(registry, new Uint8Array([0]), 8).toNumber()
+    ).toEqual(0);
+
+    expect(
+      new Int(registry, new Uint8Array([0, 0]), 16).toNumber()
+    ).toEqual(0);
+
```
Comment (jacogr) [inline packages/types-codec/src/base/Int.spec.ts:115]:
```suggestion
```

File: packages/types-codec/src/base/Int.spec.ts [hunk 2]
```diff
@@ -41,6 +41,108 @@ describe('Int', (): void => {
...
+    // MAX
+    expect(
+      new Int(registry, new Uint8Array([127]), 8).toNumber()
+    ).toEqual(127);
+
+    expect(
+      new Int(registry, new Uint8Array([255, 127]), 16).toNumber()
+    ).toEqual(32767);
+  });
+
```
Comment (jacogr) [inline packages/types-codec/src/base/Int.spec.ts:144]:
```suggestion
```

**Review by jacogr — APPROVED** [review]
Thank you.
